### PR TITLE
Add batch select to the roster view

### PR DESCRIPTION
**Pull request by lukenewman**
## Background
Added Carbon's batch select to the roster redesign, and also added placeholder UI for roster search and add record

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1358

## Associated PRs
N/A

## Validation Plan
* Have a roster with two or more records
* Select one, see the search bar replaced with the batch actions
* Select two, see the pluralization of "record"
* Deselect all from the toolbar
* Select all from the toolbar

## Automated Testing
- [x] All unit tests are passing.
- [ ] If applicable, unit tests have been added to cover this changeset.

## Additional Context
N/A

### CODE REVIEW

File: client/src/containers/EditRecord/WithdrawRecord/WithdrawRecord.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { ReportingPeriodField } from '../../../components/Forms/Enrollment/Fundi
 import { WithdrawRequest } from '../../../shared/payloads';
 import { useAlerts } from '../../../hooks/useAlerts';
 import { nameFormatter } from '../../../utils/formatters';
-import RosterContext from '../../../contexts/RosterContex/RosterContext';
+import RosterContext from '../../../contexts/RosterContext/RosterContext';
```
Comment (vonbearshark) [inline client/src/containers/EditRecord/WithdrawRecord/WithdrawRecord.tsx:22]: appreciate your doing this!

**Review by vonbearshark — COMMENTED** [review]
As of the new local dev changes the winged-keys submod should be moved outside the project, since it was removed from the gitignore

**Review by bamader — APPROVED** [review]
Pending the winged keys submodule move, this looks good for validation!

File: client/src/containers/Roster/Roster.scss [hunk 2]
```diff
@@ -26,3 +25,25 @@
...
+}
+
+.bx--batch-actions .bx--btn:hover {
+  background-color: #eddbe9;
+}
+
+.bx--action-list .bx--btn {
+  background-color: #eddbe9;
+  color: #005ea2;
+}
```
Comment (vonbearshark) [inline client/src/containers/Roster/Roster.scss:40]: nit: given that these are all the same styling (I think?), can we just `,` them together, like

```
.bx--batch-actions,
.bx--batch-actions .bx--btn:hover,
.bx--action-list .bx--btn {
  background-color: #eddbe9;
  color: #005ea2;
}
```
?

File: client/src/containers/Roster/Roster.tsx [hunk 3]
```diff
@@ -140,49 +155,106 @@ const Roster: React.FC = () => {
...
+                      getBatchActionProps().shouldShowBatchActions ? -1 : 0
+                    }
+                  />
+                  <Button
+                    className="add-record"
+                    tabIndex={
+                      getBatchActionProps().shouldShowBatchActions ? -1 : 0
+                    }
+                    // onClick={action('Add new row')}
+                    // size="small"
```
Comment (vonbearshark) [inline client/src/containers/Roster/Roster.tsx:207]: nit: rm

File: client/src/containers/Roster/Roster.tsx [hunk 4]
```diff
@@ -69,6 +84,9 @@ const Roster: React.FC = () => {
     }
   }, [user, userOrg, accessToken]);
 
+  /// this is a placeholder! batch actions to be implemented with their own function
```
Comment (vonbearshark) [inline client/src/containers/Roster/Roster.tsx:87]: nit: add `TODO`?

File: client/src/containers/Roster/Roster.tsx [hunk 5]
```diff
@@ -140,49 +155,106 @@ const Roster: React.FC = () => {
...
+                    Withdraw
+                  </TableBatchAction>
+                  <TableBatchAction
+                    tabIndex={
+                      getBatchActionProps().shouldShowBatchActions ? 0 : -1
+                    }
+                    onClick={batchActionClick(selectedRows)}
+                    renderIcon={Trash}
+                  >
+                    Delete record{selectedRows.length > 1 ? 's' : ''}
```
Comment (vonbearshark) [inline client/src/containers/Roster/Roster.tsx:191]: nit: we can use the `pluralize` package for this, I think?

**Review by vonbearshark — APPROVED** [review]
Minus the `winged-keys` looks great!

**Review by JordanGuinn — APPROVED** [review]
All LGTM!  Pending the other feedback, of course

**Comment by bamader** [comment]
## Validation Plan
- [ ] Search bar replaces with batch actions when one or more record is selected
- [ ] "Records" correctly pluralizes based on number of records selected
- [ ] Can "Select All" records in roster (including when some records are already selected)
- [ ] Can "Deselect All" records in roster (including when not all records are selected)

**Comment by bamader** [comment]
The styling on this is fantastic! Really great work Lucas / others! Validations were completely smooth here, so this is ready to merge in my book.

**Pull request merged**